### PR TITLE
Refs [TASK] [34632] [Handle event Weather Detail Screen]

### PR DESCRIPTION
**Pull request by huydd-2678**
## Related Tickets
- [#Task 34632: Handle event Weather Detail Screen](https://edu-redmine.sun-asterisk.vn/issues/34632)
## WHAT
- Bind data:
+Create request to openweather api to get data which parse to models
-Visibility
-Clouds
-Sys
-History
+Send and receive Weather, Main, Wind data from Current Weather Screen
+Pour data from remote data source into RecyclerView, TextViews
## Evidence (Screenshot or Video)
Screenshot_1617253794

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt [hunk 1]
```diff
@@ -1,18 +1,107 @@
...
+                )
+                textHumidity.text = "%d %s".format(
+                    currentWeather.currentTemp.humidity.toInt(),
+                    Constants.DEFAULT_PERCENT
+                )
+                textPressure.text = "%d %s".format(
+                    currentWeather.currentTemp.pressure.toInt(),
+                    Constants.DEFAULT_PRESSURE
+                )
+                textCloud.text = "%d %s".format(
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt]: hard code

Comment (huytq-2422) [reply app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt]: anh nghĩ là e nên viết theo kiểu extension của Int thì tốt hơn 
Int.withUnit(unit: String)

Comment (huydd-2678) [reply app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt]: vâng để e sửa thành extension, em tưởng kiểu format không tính là hard code

File: app/src/main/java/com/li/weatherapp/data/source/remote/HistoryRemoteDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }.execute()
+    }
+
+    private fun getHistory(lat: String, lon: String, dayQuantity: Int): History {
+        val currentTime = System.currentTimeMillis() / DateUtils.SECOND_IN_MILLIS
+        val jsonString = makeNetworkCall(
+            APIQueries.queryHistory(
+                lat,
+                lon,
+                (currentTime - Constants.SECOND_IN_DAY * dayQuantity).toString()
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/data/source/remote/HistoryRemoteDataSource.kt]:
```suggestion
             val time =   (System.currentTimeMillis() - DateUtils.DAY_IN_MILLIS * dayQuantity).div(DateUtils.SECOND_IN_MILLIS)
```

File: app/src/main/java/com/li/weatherapp/data/source/remote/HistoryRemoteDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            )
+        )
+        return JSONObject(jsonString).getJSONObject(History.CURRENT).let(::History)
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/data/source/remote/HistoryRemoteDataSource.kt]: sao không viết theo chiều xuôi mà lại viết theo chiều ngược thế ? 
```suggestion
        return History(JSONObject(jsonString).getJSONObject(History.CURRENT))
```

File: app/src/main/java/com/li/weatherapp/ui/adapters/HistoryViewHolder.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.*
+
+class HistoryViewHolder(private val view: View) : BaseViewHolder<History>(view) {
+
+    override fun bindData(item: History) {
+        view.run {
+            val shortDateFormat = SimpleDateFormat(Constants.SHORT_DATE_FORMAT, Locale.ENGLISH)
+            textDate.text = shortDateFormat.format(item.time.convertToMilli())
+            textTemperature.text = item.temp.toString()
+            textDescription.text = item.weather[0].main
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/adapters/HistoryViewHolder.kt]:
```suggestion
            textDescription.text = item.weather.first().main
```

File: app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt [hunk 5]
```diff
@@ -60,21 +62,36 @@ class CurrentWeatherFragment : BaseFragment(), CurrentWeatherForecastContact.Vie
...
+            Constants.DEFAULT_KILOMETER
+        )
+        textWindDegree.text = "%d %s".format(
+            weather.wind.degree.toInt(),
+            Constants.DEFAULT_DEGREE
+        )
+        textHumidity.text = "%d %s".format(
+            weather.currentTemp.humidity.toInt(),
+            Constants.DEFAULT_PERCENT
+        )
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt]: thử so sánh byte code  2 trường hợp này có gì khác nhau nhá =))

File: app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt [hunk 6]
```diff
@@ -1,18 +1,92 @@
...
+    override fun showHistory(historyList: List<History>) {
+        adapter.updateData(historyList)
+    }
+
+    override fun showMessage(data: Any) {
+        context?.showToast(data.toString())
+    }
+
+    private fun getDataFromCurrent() {
+        arguments?.getParcelable<CurrentWeather>(BUNDLE_WEATHER_DETAIL).let { currentWeather ->
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt]:
```suggestion
        val currentWeather = arguments?.getParcelable<CurrentWeather>(BUNDLE_WEATHER_DETAIL)?: return
        
```

File: app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt [hunk 7]
```diff
@@ -1,18 +1,92 @@
...
+            if (currentWeather != null) {
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/weatherdetail/WeatherDetailFragment.kt]: nếu đã dùng let rồi thì check null luôn ở let sao phải check thêm lần nữa thế

File: app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt [hunk 8]
```diff
@@ -60,21 +62,36 @@ class CurrentWeatherFragment : BaseFragment(), CurrentWeatherForecastContact.Vie
...
-            weather.wind.speed.toInt().toString() + Constants.DEFAULT_KILOMETER
-        textWindGusts.text =
-            weather.wind.degree.toInt().toString() + Constants.DEFAULT_KILOMETER
-        textWindDegree.text =
-            weather.currentTemp.humidity.toInt().toString() + Constants.DEFAULT_PERCENT
+        textWind.text = "%d %s".format(
+            weather.wind.speed.toInt(),
+            Constants.DEFAULT_KILOMETER
+        )
+        textWindDegree.text = "%d %s".format(
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt]: dùng luôn Int.withUnit()

File: app/src/main/java/com/li/weatherapp/utils/IntExtension.kt [hunk 9]
```diff
@@ -0,0 +1,3 @@
+package com.li.weatherapp.utils
+
+fun Int.withUnit(unit: String): String = "%d %s".format(this, unit)
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/utils/IntExtension.kt]: hard code 
vẫn phải khai báo hằng số cho format string

File: app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt (hunk 5 above)
Comment (huydd-2678) [reply app/src/main/java/com/li/weatherapp/ui/currentweather/CurrentWeatherFragment.kt]: vâng

**Pull request merged**